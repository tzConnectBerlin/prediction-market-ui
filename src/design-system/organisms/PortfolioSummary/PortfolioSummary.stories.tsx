import * as React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { PortfolioSummary, PortfolioSummaryProps } from './PortfolioSummary';

export default {
  title: 'Organism/PortfolioSummary',
  component: PortfolioSummary,
} as Meta;

const Template: Story<PortfolioSummaryProps> = (args) => <PortfolioSummary {...args} />;
const contentProps: PortfolioSummaryProps = {
  positions: [
    {
      type: 'trading',
      value: 95,
      weekly: 10,
    },
    {
      type: 'liquidity',
      value: 50,
      weekly: 10,
    },
  ],
  weekly: true,
};
export const DefaultCard = Template.bind({});
DefaultCard.args = contentProps;

export const WithoutWeeklyText = Template.bind({});
WithoutWeeklyText.args = {
  ...contentProps,
  weekly: false,
};
