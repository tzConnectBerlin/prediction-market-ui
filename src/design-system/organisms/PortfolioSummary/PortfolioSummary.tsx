import React from 'react';
import { useTranslation } from 'react-i18next';
import { Grid, useTheme } from '@material-ui/core';
import { Typography } from '../../atoms/Typography';
import { PaperWrapperStyled } from '../PortfolioTable/PortfolioTable';

export type Position = {
  type?: string;
  value: number;
  weekly?: number;
  currency?: string;
};

export interface PortfolioSummaryProps {
  positions: Position[];
  weekly?: boolean;
}

export const PortfolioSummary: React.FC<PortfolioSummaryProps> = ({
  positions,
  weekly = false,
}) => {
  const totalValue = positions.reduce(
    (prev, curr) => ({
      value: prev.value + curr.value,
      weekly: curr.weekly ? prev.weekly ?? 0 + curr.weekly : prev.weekly,
      currency: curr.currency,
    }),
    { value: 0, weekly: 0, currency: '$' },
  );
  const theme = useTheme();
  const { t } = useTranslation('portfolio');

  return (
    <PaperWrapperStyled square>
      <Grid marginBottom="3rem" container flexDirection="column">
        <Typography color={theme.palette.primary.main} marginBottom="1rem" size="h4">
          {t('total')} {weekly && t('weekly')}
        </Typography>
        <Typography size="h1">
          {totalValue.value} {totalValue.currency}
          {weekly && (
            <span
              style={{
                color:
                  totalValue.weekly ?? 1 > 0
                    ? theme.palette.success.main
                    : theme.palette.error.main,
              }}
            >
              ({totalValue.weekly ?? 1 > 0 ? '+' : '-'}
              {totalValue.weekly}%)
            </span>
          )}
        </Typography>
      </Grid>
      <Grid container justifyContent="space-between">
        {positions.map((item) => (
          <Grid item key={item.type} md={4} xs={8}>
            <Typography color={theme.palette.primary.main} marginBottom="1rem" size="h4">
              {t(item.type ?? '')} {weekly && t('weekly')}
            </Typography>
            <Typography>
              {item.value} {item.currency ?? 'PMM'}
              {weekly && (
                <span
                  style={{
                    color:
                      item.weekly ?? 1 > 0 ? theme.palette.success.main : theme.palette.error.main,
                  }}
                >
                  ({item.weekly ?? 1 > 0 ? '+' : '-'}
                  {item.weekly}%)
                </span>
              )}
            </Typography>
          </Grid>
        ))}
      </Grid>
    </PaperWrapperStyled>
  );
};
