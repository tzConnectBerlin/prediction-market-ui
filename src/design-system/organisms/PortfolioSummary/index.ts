export { PortfolioSummary, PortfolioSummary as default } from './PortfolioSummary';
export type { PortfolioSummaryProps } from './PortfolioSummary';
