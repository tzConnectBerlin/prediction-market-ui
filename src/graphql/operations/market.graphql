subscription Market {
  markets: allPmmStorageMarketMapOrderedsList {
    id
    marketId: idxMarketsNat
    metadataIpfsHash
    metadataDescription
    metadataAdjudicator
    marketsState
    txContext: publicTxContextByTxContextId {
      blockInfo: publicLevelByLevel {
        block: level
        bakedAt
      }
      operationGroupNumber
      operationNumber
      contentNumber
    }
    marketInfo: pmmStorageMarketMapById {
      marketBootstrapped: pmmStorageMarketMapMarketbootstrappedsByStorageMarketMapIdList {
        auctionRewardCurrencyPool: currencyPoolCreatorRewardCurrencyPool
        liquidityRewardPool: currencyPoolCreatorRewardCurrencyPool
        marketCurrencyPool: currencyPoolMarketCurrencyPool
        bootstrapYesProbability: marketBootstrappedBootstrapYesProbability
        winningPrediction: resolutionWinningPrediction
        resolutionResolvedAtBlock
        marketBootstrappedBootstrappedAtBlock
      }
      auctionRunning: pmmStorageMarketMapAuctionrunningsByStorageMarketMapIdList {
        auctionRunningAuctionPeriodEnd
        auctionRunningQuantity
        auctionRunningYesPreference
        auctionRunningUniswapContribution
      }
    }
  }
}
