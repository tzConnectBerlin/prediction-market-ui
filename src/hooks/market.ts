import { useParams } from 'react-router-dom';
import { MARKET_ADDRESS } from '../globals';

interface MarketPathParams {
  marketAddress?: string;
  questionHash?: string;
}

export const useMarketPathParams = (): MarketPathParams => {
  const { marketAddress, questionHash } = useParams<MarketPathParams>();
  const mrktAddr = marketAddress === 'main' ? MARKET_ADDRESS : marketAddress;
  return {
    marketAddress: mrktAddr,
    questionHash,
  };
};
