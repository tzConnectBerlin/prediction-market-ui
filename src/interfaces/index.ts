export * from './wallet';
export * from './router';
