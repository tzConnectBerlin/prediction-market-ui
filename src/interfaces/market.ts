import { MichelsonMap } from '@taquito/taquito';
import { BigNumber } from 'bignumber.js';

export type QuestionType = string;

export interface Links {
  label: string;
  url: string;
}

export enum MarketStateType {
  auctionRunning = 'auction',
  marketBootstrapped = 'market',
}

export enum Role {
  participant = 'Participant',
  adjudicator = 'Adjudicator',
}

export interface AuctionNode {
  auctionRunningAuctionPeriodEnd: string;
  auctionRunningQuantity: string;
  auctionRunningYesPreference: string;
  auctionRunningUniswapContribution: string;
}

export interface MarketNode {
  auctionRewardCurrencyPool: string;
  liquidityRewardPool: string;
  marketCurrencyPool: string;
  bootstrapYesProbability: string;
  liquidityRewardSupplyUpdatedAtBlock: string;
  winningPrediction?: string;
  resolutionResolvedAtBlock?: number;
  marketBootstrappedBootstrappedAtBlock: string;
}

export interface PortfolioMarket {
  question: (string | undefined)[];
  holdings: string[] | string;
  price: string[] | string;
  total: string[] | string;
}

export interface PortfolioAuction {
  question: string;
  probability: string;
  quantity: string;
}

export interface MarketStateNode {
  node: AuctionNode | MarketNode;
}

export interface StorageMarketMapMarketBootstrappeds {
  nodes: MarketStateNode[];
}

export interface StorageMarketMapAuctionRunnings {
  nodes: MarketStateNode[];
}

export interface GraphMarketNode {
  node: GraphMarket;
}
export interface GraphMarket {
  id: number;
  block: number;
  deleted: boolean;
  marketId: string;
  metadataIpfsHash: string;
  metadataDescription: string;
  metadataAdjudicator: string;
  state: string;
  dateTime: {
    bakedAt: string;
  };
  storageMarketMapAuctionRunnings: StorageMarketMapAuctionRunnings;
  storageMarketMapMarketBootstrappeds: StorageMarketMapMarketBootstrappeds;
}

export interface IPFSMarketData {
  question: QuestionType;
  auctionEndDate: string;
  iconURL?: string;
  ticker: string;
}

export interface Market extends Partial<AuctionNode>, Partial<MarketNode>, IPFSMarketData {
  marketId: string;
  ipfsHash: string;
  description: string;
  adjudicator: string;
  state: MarketStateType;
  yesPrice: number;
  liquidity?: number | string;
  block: number;
  bakedAt: string;
}

export interface AllMarkets {
  markets: {
    marketNodes: GraphMarket[];
  };
}

export interface AllLedgers {
  ledgers: {
    ledgerMaps: LedgerMap[];
  };
}

export interface AllMarketsLedgers {
  markets: {
    marketNodes: GraphMarket[];
  };
  ledgers: {
    ledgerMaps: LedgerMap[];
  };
}

export interface MarketCardStatistic {
  type: string;
  value: string | number;
  changes?: string;
  currency?: Currency;
  tokenType?: TokenType;
}

export interface MarketCardToken {
  type: TokenType;
  value: number | string;
}

export interface MarketCardData extends Market {
  tokens?: MarketCardToken[];
  statistics?: MarketCardStatistic[];
}

export interface CreateMarket {
  marketId: number;
  ipfsHash: string;
  description: string;
  adjudicator: string;
  auctionEnd: string;
  initialBid: number;
  initialContribution: number;
  tokenType: 'fa12' | 'fa2';
  tokenAddress: string;
}

export interface GraphBet {
  probability: string;
  quantity: string;
}

export interface Bet {
  block: number;
  originator: string;
  probability: number;
  quantity: number;
  marketId: string;
}

export interface BetEdge {
  bet: GraphBet;
}

export interface GraphBets {
  totalBets: number;
  betEdges: BetEdge[];
}

export interface LqtProviderNode {
  id: number;
  marketId: string;
  originator: string;
  bets: GraphBets;
  block: number;
}

export interface LqtProviderEdge {
  lqtProviderNode: LqtProviderNode;
}

export interface StorageLiquidityProviderMaps {
  lqtProviderEdge: LqtProviderEdge[];
}

export interface AllBets {
  storageLiquidityProviderMaps: StorageLiquidityProviderMaps;
}

export enum MarketTradeType {
  payIn = 'payIn',
  payOut = 'payOut',
}

export enum FormType {
  buy = 'Buy',
  sell = 'Sell',
  addLiquidity = 'Add Liquidity',
  removeLiquidity = 'Remove Liquidity',
}

export interface LiquidityValues {
  probability: number;
  quantity: number;
}

export interface TokenSupplyMap {
  id: number;
  tokenId: string;
  totalSupply: string;
  tokenReserve: string;
  deleted: boolean;
}

export interface StorageSupplyMaps {
  supplyMaps: TokenSupplyMap[];
}

export interface AllTokens {
  storageSupplyMaps: StorageSupplyMaps;
}

export enum TokenType {
  yes = 'Yes',
  no = 'No',
}

export interface LedgerMap {
  id: number;
  block: number;
  deleted: boolean;
  owner: string;
  tokenId: string;
  quantity: string;
}

export interface StorageLedgerMaps {
  ledgerMaps: LedgerMap[];
}

export interface Token {
  id: number;
  tokenId: string;
  quantity: string;
  block: number;
  dateTime: {
    bakedAt: string;
  };
}

export interface TokenQuantity {
  token: Token[];
}

export interface MarketPricePoint {
  yesPrice: number;
  block: number;
  bakedAt: string;
}

export interface AddressTokens {
  tokenQuantity: TokenQuantity;
}

export interface AuctionMarkets {
  [key: string]: Market[];
}

// TODO: clean the stuff below

export interface QuestionState {
  [key: number]: {
    [key in MarketStateType]: symbol;
  };
}

export enum Currency {
  USD = '$',
  EUR = 'â‚¬',
}

export type CurrencyTypes = keyof typeof Currency;

export enum MarketEntrypoint {
  Bid = 'bid',
  BuyYes = 'buyYes',
  BuyNo = 'buyNo',
  ClaimWinnings = 'claimWinnings',
  CloseAuction = 'closeAuction',
  CreateQuestion = 'createQuestion',
  WithdrawAuction = 'withdrawAuction',
}

export interface CreateQuestion {
  question: QuestionType;
  auctionEndDate: Date | string;
  marketCloseDate: Date | string;
  iconURL?: string;
  yesAnswer: string;
  rate?: number;
  quantity?: number;
}

export interface Bid {
  question: QuestionType;
  rate: number;
  quantity: number;
}

export interface BuyToken {
  tokenType: TokenType;
  question: QuestionType;
  quantity: number;
}

export interface CloseMarket {
  tokenType: TokenType;
  question: QuestionType;
}

export interface ClaimWinnings {
  question: QuestionType;
}

export interface QuestionMetaData extends CreateQuestion {
  hash: string;
}

export interface BidEntry {
  rate: number;
  quantity: number;
  total_token: number;
}

export type BidRegistry = MichelsonMap<string, BidEntry>;

export interface QuestionEntry {
  owner: string;
  state: QuestionState;
  auction_end: string;
  market_close: string;
  auction_bids: BidRegistry;
  uniswap_pool?: unknown;
  tokens: {
    yes_token_id: BigNumber;
    no_token_id: BigNumber;
    lqt_token_id: BigNumber;
  };
  answer?: string | null;
  winning_token?: BigNumber | null; // token that will receive the reward
  total_auction_quantity: BigNumber;
  yes_preference: BigNumber;
  uniswap_contribution_factor: BigNumber;
}

export interface AuctionData {
  yes: number;
  no: number;
  participants: number;
}

export interface AuctionDataMap {
  [key: string]: AuctionData;
}

export interface MarketFilter {
  auctions: boolean;
  allMarkets: boolean;
  openMarkets: boolean;
  closedMarkets: boolean;
  onlyMyMarkets: boolean;
}

/**
 * Temp middleware interfaces
 */

export interface BidEntryMDW {
  rate: string;
  quantity: string;
  total_token: string;
}

export interface BidRegistryMDW {
  [key: string]: BidEntryMDW;
}

export interface QuestionEntryMDW {
  owner: string;
  state: {
    [key in MarketStateType]?: null;
  };
  auction_end: string;
  market_close: string;
  auction_bids: BidRegistryMDW;
  uniswap_pool?: unknown;
  tokens: {
    yes_token_id: string;
    no_token_id: string;
    lqt_token_id: string;
  };
  answer?: string | null;
  winning_token?: string | null;
  total_auction_quantity: string;
  yes_preference: string;
  uniswap_contribution_factor: string;
  price_yes?: string;
}

export interface QuestionEntryMDWMap {
  [key: string]: QuestionEntryMDW;
}

export interface StableCoinResponse {
  [key: string]: string;
}

export interface LedgerBalanceResponse {
  [tokenId: string]: { [address: string]: string };
}

export interface ContractError {
  [key: number]: string;
}

export interface DropDownItems {
  label: string;
  value: string | number;
}
