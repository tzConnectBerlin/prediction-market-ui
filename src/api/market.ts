import { getQuestionData } from '../contracts/Market';
import { AuctionData, AuctionDataMap, CreateQuestion, QuestionMetaData } from '../interfaces';
import { fetchIPFSData } from '../ipfs/ipfs';
import { divideDown } from '../utils/math';

export const getIPFSDataByKeys = async (questionKeys: string[]): Promise<QuestionMetaData[]> => {
  const data: QuestionMetaData[] = await Promise.all(
    questionKeys.map(async (hash) => {
      const questionData = await fetchIPFSData<CreateQuestion>(hash);
      return { hash, ...questionData };
    }),
  );
  return data;
};

export const getAuctionData = async (questionHashes: string[]): Promise<AuctionDataMap> => {
  const questionDetails = await getQuestionData(questionHashes);
  return Object.keys(questionDetails).reduce((acc: AuctionDataMap, hash: string) => {
    const item = questionDetails[hash];
    const yesVal = item.yes_preference.dividedToIntegerBy(item.total_auction_quantity).toNumber();
    let yes = divideDown(yesVal);
    if (Number.isNaN(yes)) {
      yes = 0.5;
    }
    const result: AuctionData = {
      no: Math.round((1 - yes + Number.EPSILON) * 100) / 100,
      yes: Math.round((yes + Number.EPSILON) * 100) / 100,
      participants: item.auction_bids.size,
    };
    return { ...acc, [hash]: result };
  }, {});
};
