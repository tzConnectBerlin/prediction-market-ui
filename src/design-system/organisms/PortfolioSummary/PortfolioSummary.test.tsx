import renderer from 'react-test-renderer';
import { PortfolioSummary, PortfolioSummaryProps } from './PortfolioSummary';
import { Currency } from '../../../interfaces/market';

const defaultContentProps: PortfolioSummaryProps = {
  positions: [
    {
      type: 'trading',
      value: 95,
      currency: Currency.USD,
      weekly: 10,
    },
    {
      type: 'liquidity',
      value: 50,
      currency: Currency.USD,
      weekly: 10,
    },
  ],
  weekly: true,
};
describe('Snapshot testing PortfolioSummary Component', () => {
  it('renders correctly with weekly data', () => {
    const Card = renderer.create(<PortfolioSummary {...defaultContentProps} />).toJSON();
    expect(Card).toMatchSnapshot();
  });
});
