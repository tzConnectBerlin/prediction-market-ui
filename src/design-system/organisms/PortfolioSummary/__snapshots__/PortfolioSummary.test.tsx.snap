// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing PortfolioSummary Component renders correctly with weekly data 1`] = `
.emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  padding: 2rem;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-bottom: 3rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 2.125rem;
  line-height: 1.235;
  letter-spacing: 0.00735em;
  color: #1976d2;
  margin-bottom: 1rem;
}

.emotion-2.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-3 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 300;
  font-size: 6rem;
  line-height: 1.167;
  letter-spacing: -0.01562em;
}

.emotion-3.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-4 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

@media (min-width:600px) {
  .emotion-5 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 50%;
  }
}

@media (min-width:900px) {
  .emotion-5 {
    -webkit-flex-basis: 33.333333%;
    -ms-flex-preferred-size: 33.333333%;
    flex-basis: 33.333333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 33.333333%;
  }
}

.emotion-7 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
}

.emotion-7.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation1 emotion-0"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined emotion-1"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4  emotion-2"
      lines={3}
    >
      total
       
      weekly
    </h4>
    <h1
      className="MuiTypography-root MuiTypography-h1  emotion-3"
      lines={3}
    >
      145
       
      $
      <span
        style={
          Object {
            "color": "#d32f2f",
          }
        }
      >
        (
        -
        0
        %)
      </span>
    </h1>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined emotion-4"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-6 MuiGrid-grid-md-4 emotion-5"
    >
      <h4
        className="MuiTypography-root MuiTypography-h4  emotion-2"
        lines={3}
      >
        trading
         
        weekly
      </h4>
      <p
        className="MuiTypography-root MuiTypography-body1  emotion-7"
        lines={3}
      >
        95
         
        $
        <span
          style={
            Object {
              "color": "#2e7d32",
            }
          }
        >
          (
          +
          10
          %)
        </span>
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-6 MuiGrid-grid-md-4 emotion-5"
    >
      <h4
        className="MuiTypography-root MuiTypography-h4  emotion-2"
        lines={3}
      >
        liquidity
         
        weekly
      </h4>
      <p
        className="MuiTypography-root MuiTypography-body1  emotion-7"
        lines={3}
      >
        50
         
        $
        <span
          style={
            Object {
              "color": "#2e7d32",
            }
          }
        >
          (
          +
          10
          %)
        </span>
      </p>
    </div>
  </div>
</div>
`;
